pennylane/devices/device_api.py

```diff
@@ -871,3 +958,20 @@ def eval_jaxpr(
 
         """
         raise NotImplementedError
+
+
+def _default_mcm_method(capabilities: DeviceCapabilities, shots_present: bool) -> str:
+    """Simple strategy to find the best match for the default mcm method."""
+
+    supports_one_shot = "one-shot" in capabilities.supported_mcm_methods
+    has_device_support = "device" in capabilities.supported_mcm_methods
+
+    # In finite shots mode, the one-shot method is the default even if there is device support.
+    # This is to ensure consistency with old behaviour. Although I'm not too sure about this.
```

Comment:
Old behavior didn't even have `"device"` support. Are there any examples of a "device" mcm method? Would tree traversal be considered `"device"`?

Reply:
The decision was that `"device"` is an umbrella term for whatever a device does to support the full range of MCM features including post-selection. If a device has full `"device"` support for MCM, and `mcm_method="device"` is requested, we don't apply any transforms in preprocessing and trusts that the device can execute the circuit properly. What the device does under the hood when `"device"` is requested is up to the device. In that sense, tree-traversal could be considered `"device"`, if we make it so that `default.qubit` does tree-traversal when it sees `mcm_method="device"` in the execution config.